pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -32,168 +46,173 @@ abstract class BuildConfig {
   /// dependency of another. For this it is convenient to know the packageRoot.
   Uri get packageRoot;
 
-  /// The target being compiled for.
-  ///
-  /// Not available in [dryRun].
-  Target get target;
-
   /// The architecture being compiled for.
   ///
-  /// Not available in [dryRun].
-  Architecture get targetArchitecture;
+  /// Not specified (`null`) during a [dryRun].
+  Architecture? get targetArchitecture;
```

Comment:
Also consider making this non-nullable, and throw if read during [dryRun].

Reply:
That's what we had at first, but it doesn't enable writing code that will both work in dry run and builds.

Constructing assets will look like:

```dart
SomeAsset(
    os: config.os
    architecture: config.dryRun ? null : config.architecture
)
```

Having it nullable removes the `config.dryRun ? null :`.